sdk/core/azure-core/azure/core/pipeline/policies/_redirect.py

```diff
@@ -152,12 +163,15 @@ def send(self, request):
         """
         retryable = True
         redirect_settings = self.configure_redirects(request.context.options)
+        original_domain = get_domain(request.http_request.url) if self.allow else None
```

Comment:
I think this needs to allow for per-request `allow` override:
```suggestion
        original_domain = get_domain(request.http_request.url) if redirect_settings["allow"] else None
```